src/fs.rs

```diff
@@ -27,6 +28,73 @@ use httprequest::HttpRequest;
 use httpresponse::HttpResponse;
 use param::FromParam;
 use server::settings::DEFAULT_CPUPOOL;
+use header::{ContentDisposition, DispositionParam, DispositionType};
+
+///Describes `StaticFiles` configiration
+///
+///To configure actix's static resources you need
+///to define own configiration type and implement any method
+///you wish to customize.
+///As trait implements reasonable defaults for Actix.
+///
+///## Example
+///
+///```rust
+/// extern crate mime;
+/// extern crate actix_web;
+/// use actix_web::http::header::DispositionType;
+/// use actix_web::fs::{StaticFileConfig, NamedFile};
+///
+/// #[derive(Default)]
+/// struct MyConfig;
+///
+/// impl StaticFileConfig for MyConfig {
+///     fn content_disposition_map(typ: mime::Name) -> DispositionType {
+///         DispositionType::Attachment
+///     }
+/// }
+///
+/// let file = NamedFile::open_with_config("foo.txt", MyConfig);
+///```
+pub trait StaticFileConfig: Default {
```

Comment:
why do you need `Default` if trait contains only static methods? that is actually more general question. maybe config should be stateful?

Reply:
It is more or less for making it possible to pass config as argument so that in future we  could make it stateful if needed.
But I can remove it as currently I don't really see the need for it to be stateful

Reply:
It is also more pretty than putting type parameter